[website] Update MultiKueue Dispatcher API site (#6271)
kubernetes-sigs/kueue: site/content/en/docs/concepts/multikueue.md

   - The manager performs a final status sync.
   - It then deletes the corresponding objects from the worker cluster.

## Workload Dispatching

> The MultiKueue Dispatcher mechanism is available since Kueue v0.13.

Author: Done.

It provides a flexible way to control how workloads are distributed across worker clusters
by allowing users to choose between built-in dispatching algorithms or implement custom ones.
This mechanism ensures efficient workload placement while minimizing resource contention and preemptions across clusters.

The `status.nominatedClusterNames` field lists the worker clusters currently being considered for scheduling the Workload,
as determined by the dispatching algorithm, and is updated while the Workload is pending admission.

The `status.clusterName` field specifies the worker cluster where the Workload has been successfully admitted,
once the field is set it becomes immutable and it's no longer possible to set `status.nominatedClusterNames`.

### AllAtOnce (Default Mode):
In this mode, the Workload is copied to all available worker clusters as soon as it obtains a QuotaReservation in the manager cluster.
This approach ensures the fastest possible admission by allowing all clusters to compete for the Workload simultaneously.

> ### Incremental:

Reviewer: Could you mention about the all worker clusters are sorted with dictionary order, and then every 3 clusters are nominated from the sorted workers?

Author: Done.

This mode introduces a gradual dispatching strategy where clusters are nominated in rounds.
Initially, a subset of up to 3 clusters is selected, and the Workload is copied only to these clusters.
If none of the nominated clusters admit the Workload within a fixed duration (5 minutes),
additional 3 clusters are incrementally added in subsequent rounds.

> ### External (Custom implementation):

Reviewer: Could you mention about needing `kueue-admission` field manager to patch .status.nominatedClusterNames as a known limitation?

Reviewer 2: Yes, for sure. Thank you for pointing this limitation out.

Author: Done.

This approach ensures the fastest possible admission by allowing all clusters to compete for the Workload simultaneously.

### Incremental:
This mode introduces a gradual dispatching strategy where clusters are nominated in rounds.
Initially, all worker clusters are sorted in dictionary order, and a subset of up to 3 clusters is selected from the sorted list.
The Workload is copied only to these nominated clusters.
If none of the nominated clusters admit the Workload within a fixed duration (5 minutes),
an additional up to 3 clusters are incrementally added in subsequent rounds, following the same dictionary order.

### External (Custom implementation):
In this mode, the selection of worker clusters is delegated to an external controller.
The external controller is responsible for setting the `.status.nominatedClusterNames` field in the Workload to specify the clusters where it should be copied.

The MultiKueue Workload Controller synchronizes the Workload with the nominated clusters.

Known Limitation:
{{% alert title="Warning" color="primary" %}}
For the external controller to patch the `.status.nominatedClusterNames` field, it must use the kueue-admission field manager.

> This is required because the kueue-admission field manager is responsible for managing updates to the `.status.nominatedClusterNames` field.
> Without this, the external controller may encounter conflicts or be unable to update the field.

Reviewer 2: ```suggestion
This is required because the kueue-admission field manager is responsible for managing updates to the `.status.nominatedClusterNames` field.
Without this, the Kueue is not able to admit the MultiKueue workloads.
```
The previous proposal seems missing the point, unless I'm missing something, let me know @mszadkow 

My understanding the fact that external controller may conflict is irrelevant. The issue is that when it takes the field ownership to say "my-field", then the core Kueue cannot admit the workload, because it  cannot reset `.status.nominatedClusterNames`, as a result it cannot set `status.clusterName` (which is done in the same request, because they are mutually exclusive). As a result it cannot set AC Ready which is needed to admit.
